web/modeltr/connection.py

```diff
@@ -65,20 +68,44 @@ def __wait_for_completion(cls, client):
             if state == psycopg2.extensions.POLL_OK:
                 break
             elif state == psycopg2.extensions.POLL_WRITE:
-                select.select([], [client.fileno()], [])
+                # select.select([], [client.fileno()], [])
+                cls.__poll_write_async_wait(client.fileno())
             elif state == psycopg2.extensions.POLL_READ:
-                select.select([client.fileno()], [], [])
+                # select.select([client.fileno()], [], [])
+                cls.__poll_read_async_wait(client.fileno())
             else:
                 raise psycopg2.OperationalError(f'__wait_for_completion->poll() returned {state}')
 
+    @classmethod
+    def __poll_write_async_wait(cls, fileno):
+        while True:
+            [arr1, arr2, arr3] = select.select([], [fileno], [])
+            if arr2 == [fileno]:
+                break
+            asyncio.run(asyncio.sleep(0.01))
```

Comment:
What is the motivation for a sleep here?

Reply:
to act in a non-blocking manner